data/transactions/logic/eval.go

```diff
@@ -3976,6 +4195,15 @@ func opTxSubmit(cx *EvalContext) {
 		*cx.FeeCredit = basics.AddSaturate(*cx.FeeCredit, overpay)
 	}
 
+	cx.appStack = append(cx.appStack, cx.Ledger.ApplicationID()) // push app
+
+	// All subtxns must have zero'd GroupID now, since it can't be set
```

Comment:
why is the comment? We do not check/enforce this

Reply:
It is enforced because there is no GroupId field for `itxn_field`.  So it is not settable in an inner txn.